src/main/scala/com.snowplowanalytics.snowplow.micro/SqliteStorage.scala

```diff
@@ -200,37 +205,72 @@ private[micro] class SqliteStorage(xa: Transactor[IO], maxEvents: Option[Int]) e
     // the UI only uses 50 currently
     val safePageSize = Math.max(Math.min(request.pageSize, 100), 1)
 
-    // Single query with window function to get both data and total count
+    // Split into two queries for better performance:
+    // 1. Count query: Fast for indexed columns, approximate for JSON columns
+    // 2. Data query: Processes only the requested page (e.g., 50 rows) instead of all rows
     val offset = (request.page - 1) * safePageSize
-    val query = fr"SELECT event_json, COUNT(*) OVER() as total_count FROM events" ++
+
+    val hasJsonFilter = request.filters.exists { filter =>
```

Comment:
Getting the full count of events with json_extract causes a full table scan operation which is super slow over large datasets. This change makes the number of rows + page count become approximate (think infinite scroll style). Stylistically I have made a minimal change to show this in the UI but can be tweaked further.